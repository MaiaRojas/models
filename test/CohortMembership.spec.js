const mongoose = require('mongoose');
const { CohortMembership, Cohort, User } = require('../')(mongoose);


describe('CohortMembership', () => {
  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__, { useNewUrlParser: true });
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  beforeEach(async () => {
    await mongoose.connection.db.dropDatabase();
    await CohortMembership.createIndexes();
    await Cohort.createIndexes();
    await User.createIndexes();
  });

  it('should ...', () => {
    const cohort = new Cohort({});
    const user = new User({});
    const cohortMembership = new CohortMembership({
      cohort: cohort._id,
      user: user._id,
      role: 'student',
    });
    return cohortMembership.save()
      .then((result) => {
        expect(result.cohort).toBe(cohort._id);
        expect(result.user).toBe(user._id);
        expect(result.role).toBe('student');
      });
  });
});
