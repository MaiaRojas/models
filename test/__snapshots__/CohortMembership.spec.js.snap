// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CohortMembership should fail validation with a wrong membership role 1`] = `
Object {
  "role": [ValidatorError: \`other\` is not a valid enum value for path \`role\`.],
}
`;

exports[`CohortMembership should fail when cohort or user are not ObjecIds 1`] = `
Object {
  "cohort": [CastError: Cast to ObjectID failed for value "cohort._id" at path "cohort"],
  "user": [CastError: Cast to ObjectID failed for value "user._id" at path "user"],
}
`;

exports[`CohortMembership should fail when missing props 1`] = `
Object {
  "cohort": [ValidatorError: Path \`cohort\` is required.],
  "role": [ValidatorError: Path \`role\` is required.],
  "user": [ValidatorError: Path \`user\` is required.],
}
`;

exports[`CohortMembership should fail with missing role 1`] = `
Object {
  "role": [ValidatorError: Path \`role\` is required.],
}
`;
